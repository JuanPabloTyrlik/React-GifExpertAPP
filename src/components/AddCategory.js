import React, { useState } from 'react';
import PropTypes from 'prop-types';

const AddCategory = ({ setCategories }) => {
    const [value, setValue] = useState('');

    const handleChange = (e) => {
        setValue(e.target.value);
    };

    const handleSubmit = (e) => {
        e.preventDefault();
        if (value.trim().length > 0) {
            setCategories((categories) => [value, ...categories]);
            setValue('');
        }
    };

    return (
        <form onSubmit={handleSubmit}>
            <input
                type="text"
                onChange={handleChange}
                value={value}
                placeholder="Shingeki no Kyojin"
            />
        </form>
    );
};

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired,
};

export default AddCategory;
