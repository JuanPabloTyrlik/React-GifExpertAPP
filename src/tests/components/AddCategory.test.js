import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import AddCategory from '../../components/AddCategory';

describe('Tests on AddCategory Component', () => {
    const setCategories = jest.fn();
    let wrapper = shallow(<AddCategory setCategories={setCategories} />);
    beforeEach(() => {
        jest.clearAllMocks();
        wrapper = shallow(<AddCategory setCategories={setCategories} />);
    });
    test('should render correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });
    test('should handle changes on Input', () => {
        let input = wrapper.find('input');
        const value = 'HelloWorld';
        input.simulate('change', { target: { value } });
        input = wrapper.find('input');
        expect(input.prop('value')).toBe(value);
    });
    test('should not call setCategories onSubmit if InputValue is empty', () => {
        wrapper.find('form').simulate('submit', { preventDefault() {} });
        expect(setCategories).not.toBeCalled();
    });
    test('should call setCategories and clean Input', () => {
        const value = 'Omae wa mou shindeiru... NANI?!';
        wrapper.find('input').simulate('change', { target: { value } });
        expect(wrapper.find('input').prop('value')).toBe(value);
        wrapper.find('form').simulate('submit', { preventDefault() {} });
        expect(setCategories).toBeCalledTimes(1);
        expect(setCategories).toBeCalledWith(expect.any(Function));
        expect(wrapper.find('input').prop('value')).toBe('');
    });
});
