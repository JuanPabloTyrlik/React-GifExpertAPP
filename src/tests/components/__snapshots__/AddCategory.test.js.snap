// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on AddCategory Component should render correctly 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    placeholder="Shingeki no Kyojin"
    type="text"
    value=""
  />
</form>
`;
